values_test.go

```diff
@@ -9,6 +9,123 @@ import (
+		"Uint":          StringTestCase{value: NewUInt(10), expected: "10"},
+		"Uint8":         StringTestCase{value: NewUInt8(8), expected: "8"},
+		"Uint16":        StringTestCase{value: NewUInt16(16), expected: "16"},
+		"Uint32":        StringTestCase{value: NewUInt32(32), expected: "32"},
+		"Uint64":        StringTestCase{value: NewUInt64(64), expected: "64"},
+		"Uint128":       StringTestCase{value: NewUInt128(128), expected: "128"},
+		"Uint256":       StringTestCase{value: NewUInt256(256), expected: "256"},
+		"int8":          StringTestCase{value: NewInt8(-8), expected: "-8"},
+		"int16":         StringTestCase{value: NewInt16(-16), expected: "-16"},
+		"int32":         StringTestCase{value: NewInt32(-32), expected: "-32"},
+		"int64":         StringTestCase{value: NewInt64(-64), expected: "-64"},
+		"int128":        StringTestCase{value: NewInt128(-128), expected: "-128"},
+		"int256":        StringTestCase{value: NewInt256(-256), expected: "-256"},
+		"word8":         StringTestCase{value: NewWord8(8), expected: "8"},
+		"word16":        StringTestCase{value: NewWord8(16), expected: "16"},
+		"word32":        StringTestCase{value: NewWord8(32), expected: "32"},
+		"word64":        StringTestCase{value: NewWord8(64), expected: "64"},
+		"ufix":          StringTestCase{value: ufix, expected: "64.01000000"},
+		"fix":           StringTestCase{value: fix, expected: "-32.11000000"},
+		"void":          StringTestCase{value: NewVoid(), expected: "()"},
+		"true":          StringTestCase{value: NewBool(true), expected: "true"},
+		"false":         StringTestCase{value: NewBool(true), expected: "true"},
+		"optionalUfix":  StringTestCase{value: NewOptional(ufix), expected: "64.01000000"},
+		"OptionalEmpty": StringTestCase{value: NewOptional(nil), expected: "nil"},
+		"string":        StringTestCase{value: NewString("Flow ridah!"), expected: "Flow ridah!"}, //TODO: Not sure i agree that this should be escaped here
+		"array":         StringTestCase{value: array, expected: "[10, TEST]"},
+		"dictionary":    StringTestCase{value: NewDictionary([]KeyValuePair{kv}), expected: "{key: value}"},
+		"bytes":         StringTestCase{value: NewBytes([]byte("foo")), expected: "foo"},
+		"address":       StringTestCase{value: NewAddress([8]byte{0, 0, 0, 0, 0, 0, 0, 1}), expected: "0000000000000001"},
+		"struct":        StringTestCase{value: fooStruct, expected: "Foo:{y: bar}"},
+		"resource":      StringTestCase{value: resource, expected: "FooResource:{bar: 1}"},
+		"event":         StringTestCase{value: event, expected: "FooEvent:{a: 1, b: foo}"},
+		"link":          StringTestCase{value: link, expected: "Link(type: Bar, targetPath: /storage/foo)"},
+		"path":          StringTestCase{value: path, expected: "/storage/foo"},
+		"type":          StringTestCase{value: TypeValue{StaticType: "Int"}, expected: "Type<Int>"},
+		"capability":    StringTestCase{value: cap, expected: "Capability<Int>(/0000000102030405/storage/foo)"},
+	}
+
```

Comment:
It would be good to iterate over all names in `sema.baseTypes` and checking that there exists a test case for it. This would prevent the case in the future where a new type/value is added, but the author forgets to add a test case for it here. This might require refactoring the tests, e.g. grouping them by type.

Reply:
Good call.

Reply:
Can you have a look at how i tried solving this in the two latest commits?

Reply:
When working on the refactoring I realized that the check actually isn't really possible, because not all base types have corresponding values in the `cadence` package. I removed the check, it's OK if we manually ensure tests for now